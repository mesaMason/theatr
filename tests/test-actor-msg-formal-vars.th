dolphin(int weight, int length):
    receive:
        eat(int num, int hours):
            print("eat()'s formal num = ")
            print(num)
            print("eat()'s formal hours = ")
            print(hours)
        swim(int num):
            print("swim()'s formal num = ")
            print(num)
    drop:
        print("inside drop()")
    after:
        int time

func main() -> int:
    print("Hello world!")
    actor d = new dolphin(5,10)
    
    // Tests that all formals are accessible
    // hello(5) | d
    // eat(40,2) | d
    // swim(30) | d
    // die() | d


    return 0
