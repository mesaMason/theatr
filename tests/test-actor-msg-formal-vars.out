Hello world!
inside drop()
eat()'s formal num = 
40
eat()'s formal hours = 
2
swim()'s formal num = 
30
